### ISSUE TITLE
Debugger: slow when stepping around in large bundled files with source maps

### ISSUE THREAD

**Issue by thetrompf** [microsoft/vscode#192182]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.81.1 6c3e3dba23e8fadc360aed75ce363ba185c49794 x64
- OS Version: Ubuntu 22.04.3 LTS x64

**Computer specs**
| HW | Description |
| :- | :- |
| CPU | AMD Ryzen 9 5950X 16-Core Processor |
| RAM | 64GB 3600 MHz |
| NVMe | 1TB PCIe 3rd gen |

Steps to Reproduce:

1. Attach debugger to large typescript project that is bundled into a single javascript file (~680k lines) with source maps.
2. Set a break point and hit it. (The first break is fast)
3. Step over once. (All stepping and local variable resolution is slow).

Launch/Debugger trace log
[vscode-debugadapter-9d89a375.json.gz](https://github.com/microsoft/vscode/files/12520604/vscode-debugadapter-9d89a375.json.gz)

<details>
<summary>Launch.json</summary>

```json5
// launch.json
{
	"version": "0.2.0",
	"configurations": [
		{
			"type": "node",
			"request": "attach",
			"name": "Attach to GraphQL server",
			"address": "localhost",
			"cwd": "${workspaceFolder}",
			"port": 9229,
			"localRoot": "${workspaceFolder}",
			"sourceMaps": true,
			"outFiles": ["${workspaceFolder}/dist/**/*.js"],
			"skipFiles": [
				"<node_internals>/**",
				"${workspaceFolder}/node_modules",
				"${workspaceFolder}/dist",
			]
		},
	],
}
```
</details>

Animation of the debugging experience with the `--disable-extensions` flag.
![Peek 2023-09-05 10-06](https://github.com/microsoft/vscode/assets/1430118/c4bd1cb7-a4c0-40c3-8744-d5661b7f8064)

If I disable sourcemaps and try to step around the bundled output (the ~680k line file), vscode becomes unresponsive and I have to exit vscode due to the OS unresponsive window prompt.
![image](https://github.com/microsoft/vscode/assets/1430118/486fb352-7f39-49b5-8478-e50810219d96)

**Comment by connor4312** [comment]
Thanks for the log. I can make a good guess about what's happening -- it looks like you have 14,000 in-scope variables due to your bundle. To confirm where time is being spent, could you grab an extension host performance profile? Instructions here https://www.nicoespeon.com/en/2019/11/fix-vscode-extension-performance-issue/#record-extension-profile

The editor being unresponsive is a different issue than the debugger, I suggest grabbing a profile of the renderer too and including that in a new issue (or I can open one for you when you receive it) https://github.com/microsoft/vscode/wiki/Performance-Issues#profiling-the-renderer-process

**Comment by thetrompf** [comment]
The CPU profile of the Extension host as requested.

[CPU-20230907T075209.767Z.cpuprofile](https://github.com/microsoft/vscode/files/12546237/CPU-20230907T075209.767Z.cpuprofile.txt)

**Comment by VSCodeTriageBot** [comment]
Hey @connor4312, this issue might need further attention.

@thetrompf, you can help us out by closing this issue if the problem no longer exists, or adding more information.

**Comment by thetrompf** [comment]
I have tried providing the needed logs. Do you need something else/more?

**Comment by connor4312** [comment]
Hi, thanks for following up 🙂 

The month after this [I reworked](https://github.com/microsoft/vscode-js-debug/pull/1827) how renames are calculated. While this does add a comparatively expensive precomputation step to figure out scopes, this is done asynchronously, and then the rename is looked up in a tree which _should_ be much better than the iterative approach we took before.

Are you continuing to run into this problem in your projects?
